### PR TITLE
finishing up basic get requests

### PR DESCRIPTION
**Pull request by justinkang01**
### Summary 

This pull request is the first step towards implementing the basic api endpoints for the Students

- [x] implemented get all students
- [x] implemented get student by ID
- [x] implemented get favorite clubs of a student
### Test Plan 

Accessed endpoints on localhost to check for proper responses for each get request.

### Breaking Changes  

<!-- - Database schema change (anything that changes DynamoDB document structure)
<!-- - Other change that could cause problems (Detailed in notes)

### CODE REVIEW

File: backend/router/student.ts [hunk 1]
```diff
@@ -1,9 +1,50 @@
...
+  } else {
+    const favorites = await student.get('favorites');
+    const favoritesArray: ClubType[] = [];
+    favorites.forEach((fav) => {
+      console.log(fav);
+      favoritesArray.push(fav);
+      res.status(200).send(fav);
+    });
+  }
 });
```
Comment (aaronkang7) [inline backend/router/student.ts]: Great job on making the endpoints! But, I'm sorry if it wasn't clear in the task description, but the getfavorites endpoint should return and array with whole club objects inside of them! Other than that, great job :). Also, make sure to pull changes in from the origin before merging!

**Review by aaronkang7 — DISMISSED** [review]

**Review by JustKong13 — DISMISSED** [review]
I like the reorganization of the file and how everything's a bit more consistent now. Everything works for me!

**Comment by justinkang01** [comment]
the changes i've made are mostly to the student.ts file. When getting a student from the endpoint, i originally made it so that only the names of the clubs are being sent instead of the club objects. this new code uses firestore document references to reference the clubs so that a student's favorites as well as their managed clubs all return objects of clubtype (using the actual club documents in the clubs collection). This means that we can edit the clubs.ts file to do a similar thing where the registedBy field can be a reference to a student document instead of just a name. another thing to mention is that maybe we can add validate functions in a utils.tsx file to validate if a club or student exists. idk how often we might use it but it could be something to make our code more modular

**Review by 20austinw — APPROVED** [review]
Looks pretty good!

**Pull request merged**